bridge/src/main/java/com/bitwarden/bridge/util/EncryptionUtils.kt

```diff
@@ -0,0 +1,207 @@
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+/**
+ * Generate a symmetric [SecretKey] that will used for encrypting IPC traffic.
+ *
+ * This is intended to be used for implementing [IBridgeService.getSymmetricEncryptionKeyData].
+ */
+fun generateSecretKey(): Result<SecretKey> = runCatching {
+    val keygen = KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES)
+    keygen.init(256)
+    keygen.generateKey()
+}
+
+/**
+ * Generate a fingerprint for the given symmetric key.
+ *
+ * This is intended to be used for implementing
+ * [IBridgeService.checkSymmetricEncryptionKeyFingerprint], which allows callers of the service
+ * to verify that they have the correct symmetric key without actually having to send the key.
+ */
+fun SymmetricEncryptionKeyData.toFingerprint(): Result<ByteArray> = runCatching {
+    val messageDigest = MessageDigest.getInstance(KeyProperties.DIGEST_SHA256)
+    messageDigest.reset()
+    messageDigest.update(this.symmetricEncryptionKey.byteArray)
+    messageDigest.digest()
+}
+
+/**
+ * Encrypt [SharedAccountData].
+ *
+ * This is intended to be used by the main Bitwarden app during a [IBridgeService.syncAccounts] call.
+ *
+ * @param symmetricEncryptionKeyData Symmetric key used for encryption.
+ */
+fun SharedAccountData.encrypt(
+    symmetricEncryptionKeyData: SymmetricEncryptionKeyData,
+): Result<EncryptedSharedAccountData> = runCatching {
+    val encodedKey = symmetricEncryptionKeyData.symmetricEncryptionKey.byteArray
+    val key = encodedKey.toSecretKey()
+    val cipher = generateCipher()
+    cipher.init(Cipher.ENCRYPT_MODE, key)
+    val jsonString = JSON.encodeToString(this.toJsonModel())
+    val encryptedJsonString = cipher.doFinal(jsonString.encodeToByteArray()).toByteArrayContainer()
+
+    EncryptedSharedAccountData(
+        initializationVector = cipher.iv.toByteArrayContainer(),
+        encryptedAccountsJson = encryptedJsonString,
+    )
+}
+
+/**
+ * Decrypt [EncryptedSharedAccountData].
+ *
+ * @param symmetricEncryptionKeyData Symmetric key used for decryption.
+ */
+internal fun EncryptedSharedAccountData.decrypt(
+    symmetricEncryptionKeyData: SymmetricEncryptionKeyData,
+): Result<SharedAccountData> = runCatching {
+    val encodedKey = symmetricEncryptionKeyData
+        .symmetricEncryptionKey
+        .byteArray
+    val key = encodedKey.toSecretKey()
+
+    val iv = IvParameterSpec(this.initializationVector.byteArray)
+    val cipher = generateCipher()
+    cipher.init(Cipher.DECRYPT_MODE, key, iv)
+    val decryptedModel = JSON.decodeFromString<SharedAccountDataJson>(
+        cipher.doFinal(this.encryptedAccountsJson.byteArray).decodeToString()
+    )
+    decryptedModel.toDomainModel()
+}
+
+/**
+ * Encrypt [AddTotpLoginItemData].
+ *
+ * This is intended to be used by the Bitwarden Authenticator app before requesting a new TOTP
```

Comment:
Still a reference to `Bitwarden Authenticator app` here.

Reply:
👍 missed this `encrypt`, good call:

b33e6d1ead8911babe05a4411b1f91a82f19b890